submissions/chris-voitova/frogger-game/app.js

```diff
@@ -0,0 +1,138 @@
+const field = {
+  width: 505,
+  height: 606,
+  edges: {
+    x: 400,
+    y: 450,
+  },
+  waterPosition: 50,
+};
+
+// Enemies our player must avoid
+const Enemy = function (x, y, speed, width = 101, height = 61) {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+  this.width = width;
+  this.height = height;
+
+  // The image/sprite for our enemies, this uses
+  // a helper we've provided to easily load images
+  this.sprite = "images/enemy-bug.png";
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+  // You should multiply any movement by the dt parameter
+  // which will ensure the game runs at the same speed for
+  // all computers.
+  this.x += this.speed * dt;
+  if (this.x > field.width) {
+    this.x = 0;
+  }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+
+const Player = function (x, y, step = 50, width = 101, height = 101) {
+  this.sprite = "images/char-boy.png";
+  this.x = x;
+  this.y = y;
+  this.width = width;
+  this.height = height;
+  this.initPosition = {
+    x: x,
+    y: y,
+  };
+  this.step = step;
+};
+
+Player.prototype.checkCollisions = function (enemies) {
+  enemies.forEach((enemy) => {
+    if (
+      enemy.x < this.x + this.width &&
+      enemy.x + enemy.width > this.x &&
+      enemy.y < this.y + this.height &&
+      enemy.y + enemy.height > this.y
+    ) {
+      alert("you are lose!");
+      this.goToInitPosition();
+    }
+  });
+};
+
+Player.prototype.goToInitPosition = function () {
+  this.x = this.initPosition.x;
+  this.y = this.initPosition.y;
+};
+
+Player.prototype.update = function () {
+  this.checkCollisions(allEnemies);
+  if (this.y <= field.waterPosition) {
+    alert("you are win!");
+    this.goToInitPosition();
+  }
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.handleInput = function (key) {
+  if (key === "up") {
```

Comment:
Mb we can use switch case here? 🤔

Reply:
I thought about that, but something went wrong 😆

Reply:
rewrited to switch case